### PR TITLE
[ML] fixing ml autoscaling decider bugs around requested node size and tier size

### PR DESCRIPTION
**Pull request by benwtrent**
Three bugs with the ML autoscaler service are corrected in this PR:

- The XContent serialized information around waiting jobs is corrected in the verbose scaling reason
- Tier will never be below the requested node size
- Anomaly and analytics task state are now appropriately checked (using same logic as node load detection)

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/AutoscalingIT.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public void testMLAutoscalingCapacity() {
             .collect(Collectors.toList());
         NativeMemoryCapacity currentScale = MlAutoscalingDeciderService.currentScale(mlNodes, 30, false);
         expectedTierBytes = (long)Math.ceil(
-            (ByteSizeValue.ofMb(50_000 + BASIC_REQUIREMENT_MB + 60_000 + BASIC_REQUIREMENT_MB).getBytes()
+            (ByteSizeValue.ofMb(50_000 + BASIC_REQUIREMENT_MB + 60_000 + BASELINE_OVERHEAD_MB).getBytes()
```
Comment (benwtrent) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/AutoscalingIT.java:102]: This changed as we now take native executable overhead into account in the tier size.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java [hunk 2]
```diff
@@ -318,7 +321,7 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
         if (waitingAnalyticsJobs.isEmpty() == false || waitingAnomalyJobs.isEmpty() == false) {
             // We don't want to continue to consider a scale down if there are now waiting jobs
             resetScaleDownCoolDown();
-            noScaleResultOrRefresh(reasonBuilder, memoryTrackingStale, new AutoscalingDeciderResult(
+            return noScaleResultOrRefresh(reasonBuilder, memoryTrackingStale, new AutoscalingDeciderResult(
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java:324]: 🤦

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java [hunk 3]
```diff
@@ -354,6 +357,23 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
                 .max()
                 .orElse(0L));
 
+        // This is an exceptionally weird state
+        // Our view of the memory is stale or we have tasks where the required job memory is 0, which should be impossible
+        if (largestJob == 0L && ((dataframeAnalyticsTasks.isEmpty() || anomalyDetectionTasks.isEmpty()) == false)) {
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java:362]: The above code should make this scenario impossible. It makes no sense to have a `0` largest job and unempty tasks. I assume some bad state here and log a warning and return a no_scale.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlScalingReason.java [hunk 4]
```diff
@@ -126,7 +126,7 @@ public void writeTo(StreamOutput out) throws IOException {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         builder.field(WAITING_ANALYTICS_JOBS, waitingAnalyticsJobs);
-        builder.field(WAITING_ANOMALY_JOBS, waitingAnalyticsJobs);
+        builder.field(WAITING_ANOMALY_JOBS, waitingAnomalyJobs);
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlScalingReason.java:129]: 🤦

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**